src/index.js

```diff
@@ -1,504 +1,6 @@
-              return <InternalLink key={index} {...props} text={split.slice(2, -2)} />
-            }
-            else if (split.startsWith("[")) {
-              return <ExternalLink key={index} {...props} text={split} />
-            }
-            else {
-              return <PlainText key={index} {...props} text={split} />
-            }
-          }
-        }
-
-        var spans = splitText(this.props.text)
-        return <span>{spans.map(span(this.props))}</span>
-    }
-})
-
-
-var MarkdownText = React.createClass({
-    rawMarkup: function() {
-        var context = this.props.context
-        var wikiLinks = function () {
-            return [
-                {
-                    type:   'lang',
-                    regex:  /\[(https?:.*?) (.*?)\]/g,
-                    replace: function (match, url, linkText) {
-                        return '<a href="' + url + '" target="_blank">' + linkText + ' <img src="external-link-ltr-icon.png" /></a>'
-                    }
-                },
-                {
-                    type:   'lang',
-                    regex:  /\[\[(.*?)\]\]/g,
-                    replace: function (match, linkText) {
-                        var asSlug = function(title) {
-                            return title.replace(/\s/g, '-').replace(/[^A-Za-z0-9-]/g, '').toLowerCase()
-                        }
-                        var contextTitle = context.join(" ⇒ ")
-                        return '<a href="#" class="internal" title="' + contextTitle + '" data-page-name="' + asSlug(linkText) + '">' + linkText + '</a>'
-                    }
-                }
-            ]
-        }
-        
-        var converter = new showdown.Converter({
-            headerLevelStart: 3,
-            tasklists: true,
-            extensions: [ wikiLinks ]
-        })
-        var rawMarkup = converter.makeHtml(this.props.text)
-        return { __html: rawMarkup}
-    },
-    
-    onClick: function(e) {
-        var target = e.target
-        if (target.tagName.toLowerCase() === 'a' && target.className.toLowerCase() === 'internal') {
-            var detail = { context: this.props.context, instance: this.props.instance, title: target.getAttribute('title'), slug: target.getAttribute('data-page-name') }
-            var eventType = e.nativeEvent.shiftKey ? 'appendToLineup' : 'replaceInLineup'
-            var event = new CustomEvent(eventType, { detail: detail })
-            window.dispatchEvent(event)
-            e.preventDefault()
-        }
-    },
-
-    render: function() {
-        return <span onClick={this.onClick} dangerouslySetInnerHTML={this.rawMarkup()} />
-    }
-})
-
-var Footer = React.createClass({
-    render: function(){
-        function li (detail, index){ return (<li key={index}>{detail}</li>); }
-        return (
-          <div>
-              {this.props.details.map(li)}
-          </div>
-        );
-    }
-});
+import { Lineup } from './lineup'
 
 ReactDOM.render(
   <Lineup />,
```

Comment:
Does it make sense to build our table of plugins here and pass it into the Lineup?